symphony-bdk-core/src/main/java/com/symphony/bdk/core/auth/impl/BotAuthenticatorRsaImpl.java

```diff
@@ -48,20 +50,17 @@ public BotAuthenticatorRsaImpl(
   /**
    * {@inheritDoc}
    */
-  @Nonnull
   @Override
-  public AuthSession authenticateBot() throws AuthUnauthorizedException {
-    AuthSession authSession = new AuthSessionImpl(this);
-    authSession.refresh();
-    return authSession;
+  public @Nonnull AuthSession authenticateBot() {
+    return new AuthSessionRsaImpl(this);
   }
 
-  protected String retrieveSessionToken() throws AuthUnauthorizedException {
+  public String retrieveSessionToken() throws AuthUnauthorizedException {
```

Comment:
Why it has to be public? Class `AuthSessionRsaImpl.java` is part of the same package so it should be fine to keep it as `protected`